### PR TITLE
Add certificate rotation test for AWS

### PR DESCRIPTION
**Pull request by vrutkovs**
Similar to baremetal-based cert rotation tests this tests that AWS cluster can recover after being suspended for n days (default: 30) in order to ensure cloud provider features are still available.

Includes https://github.com/openshift/release/pull/42773

### CODE REVIEW

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-shutdown-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
Job passed, lets run it few more times to be sure

/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

File: ci-operator/step-registry/openshift/e2e/aws/cert-rotation/openshift-e2e-aws-cert-rotation-workflow.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    - ref: aws-provision-cco-manual-users-static
+    - ref: ipi-conf-manual-creds-remove-unnecessary-creds
+    - chain: ipi-install
+    - chain: aws-provision-bastionhost
+    test:
+    - ref: openshift-e2e-aws-cert-rotation
+    post:
+    - ref: aws-deprovision-stacks
+    - ref: aws-deprovision-s3buckets
+    - ref: ipi-deprovision-deprovision
```
Comment (jianlinliu) [inline ci-operator/step-registry/openshift/e2e/aws/cert-rotation/openshift-e2e-aws-cert-rotation-workflow.yaml:16]: I think also need to append `aws-deprovision-users-and-policies` step to deprovision the resources created in `aws-provision-cco-manual-users-static`

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Comment by vrutkovs** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-aws-ovn-ha-cert-rotation-suspend-30d

**Review by dinhxuanvu — APPROVED** [review]
/lgtm

**Pull request merged**